src/test/java/com/atlassian/db/replica/it/example/aurora/replica/api/SynchronousWriteConsistency.java

```diff
@@ -0,0 +1,138 @@
+package com.atlassian.db.replica.it.example.aurora.replica.api;
+
+import com.atlassian.db.replica.internal.LazyReference;
+import com.atlassian.db.replica.it.example.aurora.replica.spi.Latency;
+import com.atlassian.db.replica.spi.ConnectionProvider;
+import com.atlassian.db.replica.spi.ReplicaConsistency;
+
+import java.sql.Connection;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.function.Supplier;
+
+import static java.lang.Math.max;
+import static java.lang.Math.min;
+
+/**
+ * Waits until writes propagate to replica.
+ * It doesn't depend on a cross-JVM cache.
+ */
+
+// TODO add also DeferredSynchronousWriteConsistency implementation
+// TODO multi replica support
+public class SynchronousWriteConsistency implements ReplicaConsistency {
```

Comment:
Deferred WAW is out of the scope for the initial MultiReplica support in the lib.